daemon/response.go

```diff
@@ -332,6 +334,104 @@ func (rr *journalLineReaderSeqResponse) ServeHTTP(w http.ResponseWriter, r *http
 	rr.Close()
 }
 
+type followRequestsSeqResponse struct {
+	requestsCh chan *promptrequests.PromptRequest
+}
+
+func newFollowRequestsSeqResponse() (*followRequestsSeqResponse, chan *promptrequests.PromptRequest) {
+	resp := &followRequestsSeqResponse{
+		requestsCh: make(chan *promptrequests.PromptRequest, 16),
+	}
+	return resp, resp.requestsCh
+}
+
+func (fr *followRequestsSeqResponse) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json-seq")
+
+	flusher, hasFlusher := w.(http.Flusher)
+
+	var err error
+	writer := bufio.NewWriter(w)
+	enc := json.NewEncoder(writer)
+	for {
+		req, received := <-fr.requestsCh
+		if !received {
+			// requestsCh is closed
+			break
+		}
+		writer.WriteByte(0x1E) // RS -- see ascii(7), and RFC7464
+		if err := enc.Encode(req); err != nil {
+			break
+		}
+		if err := writer.Flush(); err != nil {
+			break
+		}
+		if hasFlusher {
+			flusher.Flush()
+		}
+	}
+
+	if err != nil && err != io.EOF {
+		writer.WriteByte('\x1E')
```

Comment:
This is in a different form to the previous use of RS - suggest using a constant here or a function to make this clearer.

Reply:
Do you mean using `writer.WriteByte('\x1E')` instead of `writer.WriteByte(0x1E)`?

Yes, I'm mangling things a bit here. The original proposal was to do something like what's on L328 (https://github.com/olivercalder/snapd/pull/15/files#diff-5eaaa1e7300deb2dc5d6519d3de0686d75bbad535ef8c6da204c74d4225fcc45R328) but Zygmunt suggested quoting might be a problem in this case (https://github.com/snapcore/snapd/pull/12898/files#r1234464568). I switched to using `json.Marshal()` and writing the resulting `[]byte` separately from the RS and newline.

I do see other uses of `writer.WriteByte(0x1E)` in this file, so it seems totally reasonable to do that as well. Is this what you had in mind?

Reply:
It just seemed unclear why there were two different method and the hard-coded number. It would be slightly easier to read if it was `writeRS()` or `writeByte(RS)`, but this is just a nit.